Tribler/community/triblerchain/database.py

```diff
@@ -172,7 +174,12 @@ def get_graph_edges(self, public_key, neighbor_level=1):
             ORDER BY (ta.traffic_a_to_b + ta.traffic_b_to_a) DESC
         """ % {"query": query, "traffic": total_traffic}
 
-        return self.execute(query, (buffer(public_key), buffer(public_key))).fetchall()
+        def get_rows(result):
+            return result.fetchall()
+
+        d = task.deferLater(reactor, 0.0, self.execute, query, (buffer(public_key), buffer(public_key)))
```

Comment:
This creates a task on the same thread as the thread this call is made on. The delay is set to 0.0, so the task is started instantly. The alternative would be to use `threads.deferToThreads`, but that won't work as the chance is high that the task will be scheduled on a different thread than the database.